### PR TITLE
[SP-1977] - Backport of PDI-13981 - Google Analytics Input Step: Dupl…

### PR DESCRIPTION
**Pull request by akhayrutdinov**
…icate entry on 1001st entry (5.4 Suite)

- fix the problem
- add test

@mattyb149, @brosander, review it please. This PR includes Matt T's fix as well my unit test, where I tried to mimic the library's behaviour
/cc @matthewtckr

### CODE REVIEW

File: plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import java.util.List;
 
 import com.google.api.services.analytics.Analytics;
+import com.google.common.annotations.VisibleForTesting;
```
Comment (dkincade) [inline plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java]: Please remove this import - we don't have a dependency on guava which defines this annotation and we don't want to add it just to make an annotation that can be replaced by a comment for it...

File: plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java [hunk 2]
```diff
@@ -230,6 +231,7 @@ public boolean init( StepMetaInterface smi, StepDataInterface sdi ) {
     return true;
   }
 
+  @VisibleForTesting
```
Comment (dkincade) [inline plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java]: Remove this line ... see comment above (just mark it protected and add a comment that it is protected vs private for testing purposes...)

File: plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java (hunk 1 above)
Comment (mattyb149) [inline plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java]: We've used this annotation quite a few other places in kettle, should this be discouraged?

**Comment by akhayrutdinov** [comment]
@dkincade, I've done it.
However, I have the same question: I've been using the annotation for quite a long time already with no complaints from reviewers. Should I give up doing it?

**Comment by dkincade** [comment]
@akhayrutdinov we will have to discuss with @pentaho-nbaker about the future of that annotation. I am not against it in general but I hate to bring in a 3rd party library just for a dependency which acts as a comment. It will depend on what else that library includes for us (or conflicts with)

**Pull request merged**